packages/eslint-plugin/src/rules/prefer-standalone.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  ASTUtils,
+  RuleFixes,
+  isNotNullOrUndefined,
+  Selectors,
+} from '@angular-eslint/utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+type DecoratorTypes = 'component' | 'directive' | 'pipe';
+export type MessageIds = 'preferStandalone';
+export const RULE_NAME = 'prefer-standalone';
+const METADATA_PROPERTY_NAME = 'standalone';
+const IS_STANDALONE = 'true';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: `Ensures component, directive and pipe \`${METADATA_PROPERTY_NAME}\` property is set to \`${IS_STANDALONE}\` in the component decorator`,
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      preferStandalone: `The {{type}} \`${METADATA_PROPERTY_NAME}\` property should be set to \`${IS_STANDALONE}\``,
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const standaloneRuleFactory =
+      (type: DecoratorTypes) => (node: TSESTree.Decorator) => {
```

Comment:
The contents of this function is the same as for [prefer-standalone-component](https://github.com/angular-eslint/angular-eslint/blob/ee4ebf2ab381164eebf889b131c15f15a93a9693/packages/eslint-plugin/src/rules/prefer-standalone-component.ts#L34), with the addition of turning it into a factory function and passing in the decorator `type` for the messageID data property.